revolv/project/stats.py

```diff
@@ -0,0 +1,124 @@
+"""
+Module for encapsulating functions related to project statistics (aka
+computing the effects of projects based on their stated kilowatt output).
+"""
+from django.db.models import Sum
+
+
+class KilowattStatsAggregatorException(Exception):
+    pass
+
+
+class KilowattStatsAggregator(object):
+    """
+    A class to abstract the calculation of different energy statistics about a
+    project, based on the kilowatt output of the project. The idea is that in order
+    to engage donors, it makes sense for RE-volv to display the impact of a solar
+    project in terms of not only power output but dollars saved per month, equivalent
+    acres of trees that could be planted, etc.
+
+    This is a utility class that takes in a kilowatt number (or, can accept instantiation
+    from a Project or a Project queryset) and makes different statistics about the
+    kilowatt power output available through calculations. Another advantage is that if
+    we need to change the calculations, the price of electricity, or anything else
+    related to the project stats, we can do so here and only here.
+    """
+    ELECTRICITY_PRICE_PER_KWH = 0.152  # dollars
+    NUM_HOURS_IN_MONTH = 730.484
+    NUM_HOURS_IN_YEAR = 8765.81
+    POUNDS_CARBON_PER_KWH = 1.845
+    KWH_PER_ACRE_OF_TREES = 5125
+    KWH_PER_GALLON_GAS = 33.41
+    # the average mpg number may need to periodically be updated, as it is
+    # prone to fast increase!
+    AVERAGE_AMERICAN_MPG = 24.1
+    # percentage of day the solar panels produce power
+    # roughly 11.5 hours a day in CA: http://aa.usno.navy.mil/cgi-bin/aa_durtablew.pl?form=1&year=2015&task=-1&state=CA&place=San+Francisco
+    SOLAR_PANEL_USE_FACTOR_PER_DAY = 11.5 / 24
```

Comment:
These are from online sources (see links in docstrings). I was thinking that I would confirm them with Andreas when we meet with him next, but wanted to get this PR up ASAP